rcl_yaml_param_parser/src/parser.c

```diff
@@ -1105,7 +1106,22 @@ static void * get_value(
   {
     errno = 0;
     endptr = NULL;
-    dval = strtod(value, &endptr);
+    const char * iter_ptr = NULL;
+    if ((0 == strcasecmp(value, ".nan")) ||
```

Comment:
@Ada-King @clalancette @fujitatomoya 

Actually, I don't think we should support `strcasecmp`, because from the `yaml` spec([1.1](https://yaml.org/type/float.html) or [1.2](https://yaml.org/spec/1.2/spec.html)), 
it only supports
```
 [-+]? ( \.inf | \.Inf | \.INF )	 tag:yaml.org,2002:float (Infinity)
  \.nan | \.NaN | \.NAN       tag:yaml.org,2002:float (Not a number)
```

Reply:
So you mean just using `strcmp` against those should be okay? Am i understanding correctly?

Reply:
@fujitatomoya 
Yes.

Reply:
that makes sense to me. thanks

Reply:
No idea if user will share the yaml configuration with other projects. So I think it's better to implement these features by following the yaml spec.